openedx/features/edly/utils.py

```diff
@@ -0,0 +1,255 @@
+        content = content.replace(relative_urls[x], absolute_urls[x])
+    return content
+
+
+def _get_relative_urls(content):
+    """
+    This function will extract the relative urls from content
+
+    :param content: String from which we have to extract the relative imports
+    :return: List of relative urls
+    """
+    pattern = r'href' '*=' '*("/[:.A-z0-9/+@-]*")'
+    try:
+        relative_urls = re.findall(pattern, content)
+        relative_urls = [url.replace('"', '') for url in relative_urls]
+    except TypeError:
+        # If new course created the old_content will be None or Empty
+        return []
+    return relative_urls
+
+
+def _create_absolute_urls(course_key, relative_urls):
+    """
+    This function will make the absolute urls of the given relative urls
+
+    :param course_key: Course key for the assets
+    :param relative_urls: list of relative urls
+    :return: list of absolute urls
+    """
+    absolute_urls = []
+    for relative_url in relative_urls:
+        if _is_static_path(relative_url):
+            absolute_url = _generate_absolute_url_from_static_path(course_key, relative_url)
+            absolute_url = absolute_url.replace('block@', 'block/')
+            absolute_urls.append(absolute_url)
+        elif _is_canonicalized_asset_path(relative_url):
+            absolute_url = relative_url.replace('block@', 'block/')
+            absolute_urls.append(absolute_url)
+        else:
+            # This url doesn't belong to the course we are not changing it.
+            relative_urls.remove(relative_url)
+    absolute_urls = ["{lms_base_url}{path}".format(lms_base_url=settings.LMS_ROOT_URL, path=path)
+                     for path in absolute_urls]
+    return relative_urls, absolute_urls
+
+
+def _is_static_path(path):
+    return path.startswith('/static/')
+
+
+def _is_canonicalized_asset_path(path):
+    return path.startswith('/asset')
+
+
+def _generate_absolute_url_from_static_path(course_key, path):
+    """
+    This function will generate the exact path of the given static path.
+
+    :param course_key: Course for which we have to create the static assets path.
+    :param path: Static path of asset.
+    :return: Absolute path of the static asset.
+    """
+    absolute_url = StaticContent.get_canonicalized_asset_path(course_key, path, "", {})
+    return absolute_url
+
+
+def get_xblock(usage_key):
+    """
+    This function will get xblock.
+
+    :param usage_key: usage_key of xblock we are getting.
+    :return: xblock
+    """
+    store = modulestore()
+    with store.bulk_operations(usage_key.course_key):
+        try:
+            return store.get_item(usage_key, depth=None)
+        except ItemNotFoundError:
+            log.error("Can't find parent xblock.")
+            return None
+        except InvalidLocationError:
+            log.error("Can't find parent/item by location.")
+            return None
+
+
+def _get_published_unit_url(xblock):
```

Comment:
On publishing a unit, whose subsection is not published yet I got this

```
------- EMAIL -------
To: edx@example.com
From: 
Subject: Your course hassan_demo_2 on localhost:8001 has been updated
Body:
Dear 

Unit: Time Unit has been updated in hassan_demo_2.

Your can see the Unit changes here: http://localhost:18000/courses/course-v1:ucp+CS1034+2014_T2/courseware/c9f161b0e13e488e9ececf5a7eef47a6/fdeff670b3dc4c46bee70bdd4ebc6407/?activate_block_id=block-v1:ucp+CS1034+2014_T2+type@vertical+block@6f27d4b69454419b80c1ad2f8c279d02

----
Thanks,
LUMS ELM training team
-------  END  -------
```

When I go to the learner side that unit is not visible to me so why am I receiving that email?

Also  change that line `Your can see ...` to `You can view ....`

Reply:
You can click on the given link the unit is published and showing at my end.

Reply:
In my case I am recieving this message.
![image](https://user-images.githubusercontent.com/42294172/67758383-64b60b80-fa5f-11e9-9084-932278312b93.png)

Reply:
Yes, the course has not started yet. Start date of the course is in future. Let me confirm the requirements in this case.

Reply:
Done

Reply:
Still getting same message.